packages/react/src/hooks/useAuth.ts

```diff
@@ -95,17 +95,16 @@ type UseAuthOptions = Record<string, any> | PendingSessionOptions | undefined |
 export const useAuth = (initialAuthStateOrOptions: UseAuthOptions = {}): UseAuthReturn => {
   useAssertWrappedByClerkProvider('useAuth');
 
-  const { treatPendingAsSignedOut, ...rest } = initialAuthStateOrOptions ?? {};
-  const initialAuthState = rest as any;
+  const { treatPendingAsSignedOut } = initialAuthStateOrOptions ?? {};
 
-  const authContextFromHook = useAuthContext();
-  let authContext = authContextFromHook;
+  const isomorphicClerk = useIsomorphicClerkContext();
 
-  if (authContext.sessionId === undefined && authContext.userId === undefined) {
-    authContext = initialAuthState != null ? initialAuthState : {};
-  }
```

Comment:
No need to see if we can trust the authContext. We can just the the authStore directly here